atomisticparsers/h5md/parser.py

```diff
@@ -0,0 +1,787 @@
+            if self.h5md_positions_group_all is None:
+                return
+            self._h5md_positions_value_all = self._data_parser.hdf5_getter(self.h5md_positions_group_all, 'value')
+        return self._h5md_positions_value_all
+
+    @property
+    def n_frames(self):
+        if self._n_frames is None:
+            if self.h5md_positions_value_all is None:
+                return
+            self._n_frames = len(self.h5md_positions_value_all) if self.h5md_positions_value_all is not None else None
+        return self._n_frames
+
+    @property
+    def n_atoms(self):
+        if self._n_atoms is None:
+            if self.h5md_positions_value_all is None:
+                return
+            self._n_atoms = [len(pos) for pos in self.h5md_positions_value_all] if self.h5md_positions_value_all is not None else None
+        return self._n_atoms
+
+    @property
+    def frame_rate(self):  # TODO extend to non-fixed n_atoms
+        if self._frame_rate is None:
+            if self._n_atoms == 0 or self._n_frames == 0:
+                self._frame_rate = 1
+            else:
+                cum_atoms = self._n_atoms * self._n_frames
+                self._frame_rate = 1 if cum_atoms <= self._cum_max_atoms else cum_atoms // self._cum_max_atoms
+        return self._frame_rate
+
+    @property
+    def atom_parameters(self):
+        if self._atom_parameters is None:
+            if not self.h5md_particle_group_all:
+                return
+            self._atom_parameters = {}
+            n_atoms = self.n_atoms[0]  # TODO Extend to non-static n_atoms
+            if n_atoms is None:
+                return
+            particles = self.h5md_particle_group_all  # TODO Extend to arbitrary particle groups
+
+            atom_parameter_keys = ['label', 'mass', 'charge']
+            for key in atom_parameter_keys:
+                value = self._data_parser.hdf5_getter(particles, self._nomad_to_particles_group_map[key])
+                if value is not None:
+                    self._atom_parameters[key] = value
+                else:
+                    continue
+                if type(self._atom_parameters[key]) == h5py.Group:
+                    self.logger.warning('Time-dependent ' + key + ' currently not supported. Atom parameter values will not be stored.')
+                    continue
+                elif len(self._atom_parameters[key]) != n_atoms:
+                    self.logger.warning('Inconsistent length of ' + key + ' . Atom parameter values will not be stored.')
+                    continue
+        return self._atom_parameters
+
+    @property
+    def system_info(self):
+        if self._system_info is None:
+            particles_group = self.h5md_particle_group_all
+            positions_group = self.h5md_positions_group_all
+            positions_value = self.h5md_positions_value_all
+            if not particles_group:
+                return
+            self._system_info = {'system': {}, 'calculation': {}}
```

Comment:
i think you should return dict(system={}...) if particles_group is None otherwise you call this everytime you access system_info

Reply:
done.